src/java/picard/analysis/directed/GcBiasMetricsCollector.java

```diff
@@ -0,0 +1,286 @@
+        private double sum(final int[] values) {
+            final int length = values.length;
+            double total = 0;
+            for (int i = 0; i < length; ++i) {
+                total += values[i];
+            }
+
+            return total;
+        }
+
+        /////////////////////////////////////////////////////////////////////////////
+        //Called to add metrics to the output file for each level of collection
+        // these metrics are used for graphing gc bias in R script
+        /////////////////////////////////////////////////////////////////////////////
+        public void addMetricsToFile(final MetricsFile<GcBiasMetrics, Integer> file) {
+            for (final Map.Entry<String, GcObject> entry : gcData.entrySet()) {
+                final GcObject gcCur = entry.getValue();
+                final String gcType = entry.getKey();
+                final int[] readsByGc = gcCur.readsByGc;
+                final long[] errorsByGc = gcCur.errorsByGc;
+                final long[] basesByGc = gcCur.basesByGc;
+                final int totalClusters = gcCur.totalClusters;
+                final int totalAlignedReads = gcCur.totalAlignedReads;
+                final String group = gcCur.group;
+                final GcBiasMetrics metrics = new GcBiasMetrics();
+                final double totalWindows = sum(windowsByGc);
+                final double totalReads = sum(readsByGc);
+                final double meanReadsPerWindow = totalReads / totalWindows;
+                if(totalAlignedReads>0) {
+                    for (int i = 0; i < windowsByGc.length; ++i) {
+                        final GcBiasDetailMetrics m = new GcBiasDetailMetrics();
+                        m.GC = i;
+                        m.WINDOWS = windowsByGc[i];
+                        m.READ_STARTS = readsByGc[i];
+                        if (errorsByGc[i] > 0) m.MEAN_BASE_QUALITY = QualityUtil.getPhredScoreFromObsAndErrors(basesByGc[i], errorsByGc[i]);
+                        if(windowsByGc[i]!=0) {
+                            m.NORMALIZED_COVERAGE = (m.READ_STARTS / (double) m.WINDOWS) / meanReadsPerWindow;
+                            m.ERROR_BAR_WIDTH = (Math.sqrt(m.READ_STARTS) / (double) m.WINDOWS) / meanReadsPerWindow;
+                        }
+                        else{
+                            m.NORMALIZED_COVERAGE = 0;
+                            m.ERROR_BAR_WIDTH = 0;
+                        }
+                        m.ACCUMULATION_LEVEL = group;
+                        if(group.equals("Read Group")) {m.READ_GROUP = gcType;}
+                        else if(group.equals("Sample")) {m.SAMPLE = gcType;}
+                        else if(group.equals("Library")) {m.LIBRARY = gcType;}
+                        metrics.DETAILS.addMetric(m);
+                    }
+                    // Synthesize the high level summary metrics
+                    final GcBiasSummaryMetrics s = new GcBiasSummaryMetrics();
```

Comment:
Even though it's only used briefly, better to use a longer/more informative name than 's' for a variable.